File: ruby/secret-handshake/secret_handshake.rb
class SecretHandshake
  attr_reader :commands

  def initialize(n)
    @commands = []

    begin
      @commands << 'wink' if n & 0b1 > 0
      @commands << 'double blink' if n & 0b10 > 0
      @commands << 'close your eyes' if n & 0b100 > 0
      @commands << 'jump' if n & 0b1000 > 0
      @commands.reverse! if n & 0b10000 > 0
    rescue NoMethodError
    end
  end
end

File: javascript/secret-handshake/secret-handshake.js
export const commands = (n) => {
    let comms = [];

    if (n & 1)      comms.push("wink");
    if (n & 1 << 1) comms.push("double blink");
    if (n & 1 << 2) comms.push("close your eyes");
    if (n & 1 << 3) comms.push("jump");
    if (n & 1 << 4) comms.reverse();

    return comms;
};
